pkg/kapp/app/recorded_app.go

```diff
@@ -80,9 +85,26 @@ func (a *RecordedApp) UpdateUsedGVs(gvs []schema.GroupVersion) error {
 func (a *RecordedApp) CreateOrUpdate(labels map[string]string) error {
```

Comment:
@praveenrewar I pretty heavily refactored this to be a bit more maintainable / small. Lemme know whatcha think!

Reply:
Looks really good now.